docs/maintenance/internals/systemfield.md

```diff
@@ -1,47 +1,701 @@
+
+```python
+from invenio_records.systemfields import SystemField
+
+class UppercaseField(SystemField):
+    """A field that automatically converts values to uppercase."""
+
+    def __get__(self, record, owner=None):
+        if record is None:
+            return self  # Class access
+
+        # Get value from record dictionary
+        value = self.get_dictkey(record)
+        return value.upper() if value else None
+
+    def __set__(self, record, value):
+        if value is not None:
+            # Store lowercase in record, display uppercase
+            self.set_dictkey(record, value.lower(), create_if_missing=True)
+
+# Usage in the your API class
+class MyRecord(Record, SystemFieldsMixin):
+    title = UppercaseField('metadata.title')
+
+record = MyRecord({'metadata': {'title': 'hello world'}})
+print(record.title)  # "HELLO WORLD"
+record.title = "New Title"
+print(record['metadata']['title'])  # "new title"
+```
+
+### Advanced Field with Lifecycle Hooks
+
+```python
+class TimestampField(SystemField):
+    """A field that tracks creation and modification times."""
+
+    def __init__(self, creation_key='created', modified_key='modified'):
+        super().__init__()
+        self.creation_key = creation_key
+        self.modified_key = modified_key
+
+    def __get__(self, record, owner=None):
+        if record is None:
+            return self
+
+        return {
+            'created': record.get(self.creation_key),
+            'modified': record.get(self.modified_key)
+        }
+
+    def __set__(self, record, value):
+        # Don't allow direct setting
+        raise AttributeError("Timestamps are managed automatically")
+
+    def post_init(self, record, data, model=None, field_data=None):
+        """Set creation timestamp on new records."""
+        from datetime import datetime
+        now = datetime.utcnow().isoformat()
+
+        if self.creation_key not in record:
+            record[self.creation_key] = now
+        record[self.modified_key] = now
+
+    def pre_commit(self, record):
+        """Update modification timestamp before saving."""
+        from datetime import datetime
+        record[self.modified_key] = datetime.utcnow().isoformat()
+
+# Usage
+class MyRecord(Record, SystemFieldsMixin):
+    timestamps = TimestampField()
+
+record = MyRecord({})  # Automatically sets created/modified
+# ... later modifications will update modified timestamp
+```
+
+## Relations
+
+Relations are specialized SystemFields that manage connections between records and external entities (other records, database models, APIs, etc.).
+
+### Basic Relation Concepts
+
+Relations consist of several components:
+
+- **RelationsField**: The main system field that manages multiple relations
+- **Relation Classes**: Define how to resolve and validate specific relation types
+- **Result Classes**: Handle the returned values when accessing relations
+- **Mapping Class**: Provides the interface for managing relations on a record
```

Comment:
Can we make explicit the relationship between these classes and the concrete code below or concrete code in general? For example, RelationsField is `invenio_records.systemfields.relations.RelationsField`, a Relation class can be `PKRelation` and so on...

`Mapping Class` -> `Mapping Classes`